stdlib/runtime/malloc.gr

```diff
@@ -172,23 +172,27 @@ export let free = (ap: WasmI32) => {
   // is actually already pointing to this node, so we don't do anything.
   if (blockPtr != freePtr) {
     // Find the location to insert this block into the free list
-    for ( ; !((blockPtr > p) && (blockPtr < getNext(p))); p = getNext(p) ) {
-      if ((p >= getNext(p)) && ((blockPtr > p) || (blockPtr < getNext(p)))) {
+    for ( ; ; ) {
```

Comment:
Would we want this to just be `while(true)`?

Reply:
Changed into while(true). Looks a bit less cryptic, but the worse offender for readability remains the condition for the break. I'd compensate with a comment, but I lack a proper understanding of the algorithm.